### PR TITLE
apptainer: new package

### PR DESCRIPTION
**Pull request by matz-e**
See spack/spack#30745, based heavily on the `singularityce` recipe.

### CODE REVIEW

**Review by DriesVerachtert — APPROVED** [review]
Looks great!

File: var/spack/repos/builtin/packages/apptainer/spack_perms_fix.sh.j2 [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#!/bin/sh -eu
+
+{% for cf  in chown_files %}
+chown root {{ prefix }}/{{ cf }}
+{% endfor %}
+
+{% for sf in setuid_files %}
+chmod 4555 {{ prefix }}/{{ sf }}
+{% endfor %}
```
Comment (pramodk) [inline var/spack/repos/builtin/packages/apptainer/spack_perms_fix.sh.j2:9]: Just to confirm - this is part of the instructions from spack build log that they ask to run?

**Review by pramodk — APPROVED** [review]
Looks reasonable to me!

File: var/spack/repos/builtin/packages/apptainer/spack_perms_fix.sh.j2 (hunk 1 above)
Comment (matz-e) [reply var/spack/repos/builtin/packages/apptainer/spack_perms_fix.sh.j2:9]: Yes, this file will be populated with stuff from the recipe and saved in `$(spack location -i apptainer)/bin`.

It's maybe easiest to just run this on updates to `apptainer` (after review)

**Pull request merged**